okapi-core/src/main/java/org/folio/okapi/managers/TimerManager.java

```diff
@@ -0,0 +1,312 @@
+package org.folio.okapi.managers;
+
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.bean.InterfaceDescriptor;
+import org.folio.okapi.bean.ModuleDescriptor;
+import org.folio.okapi.bean.ModuleInstance;
+import org.folio.okapi.bean.RoutingEntry;
+import org.folio.okapi.bean.TimerDescriptor;
+import org.folio.okapi.common.OkapiLogger;
+import org.folio.okapi.util.LockedTypedMap2;
+
+public class TimerManager {
+
+  private final Logger logger = OkapiLogger.get();
+  private static final String TIMER_ENTRY_SEP = "_";
+  private static final String MAP_NAME = "org.folio.okapi.timer.map";
+  private static final String EVENT_NAME = "org.folio.okapi.timer.event";
+  private final LockedTypedMap2<TimerDescriptor> tenantTimers
+      = new LockedTypedMap2<>(TimerDescriptor.class);
+  private final Set<String> timerRunning = new HashSet<>();
```

Comment:
Does this need to be tenant id specific?

Reply:
Yes. Will fix that